tests/unit/multiclass/tree/RandomForest_unittest.cc

```diff
@@ -272,7 +272,7 @@ TEST(RandomForest,classify_non_nominal_test)
 	EXPECT_EQ(1.0,res_vector[4]);
 
 	CMulticlassAccuracy* eval=new CMulticlassAccuracy();
-	EXPECT_NEAR(0.714285,c->get_oob_error(eval),1e-6);
+	EXPECT_NEAR(0.642857,c->get_oob_error(eval),1e-6);
 
```

Comment:
Hi @vigsterkr and @karlnapf , so the int32_t replacement finally comes to the unit test refactoring. First of all, 
I'm not sure if these changes for `RandomForest_unittes`, `KernelSelection_unittest` and  `RandomCARTree_unittest.cc` make sense to you. But I remember we had discuss these test for random refactor and we need to change them somehow. So I just want to point it out in here :)

Secondly, for CMath permute in [here](https://github.com/shogun-toolbox/shogun/pull/3960/commits/9a6c83f25bf09897cc4b13529c1388a9f74b13a7#diff-fad5669b3799897497e99f4822939b91R387), I really don't know why the result changed. Does replace int32_t with index_t can make that change happened? I think I need to look into it later. And verify its result on Linux platform(I'm working on the OSX right now)

there still have some unit tests failed locally:

```
The following tests FAILED:
	179 - unit-ProbingSampler (Failed)
	285 - integration_meta_cpp-clustering-gmm (Failed)
	294 - integration_meta_cpp-distance-cosine (Failed)
```

And for ProbingSampler, it depend on colpack and looks like doesn't aceept `index_t`.(it can only use int, I think). I'm not sure if that's the reason why make the result different, will look into it later :)